File: Algorithms/[1, 1k)/334/increasing-triplet-subsequence.c
bool increasingTriplet(int* nums, int numsSize){
    if (numsSize < 3) return false;
    
    int dp[3];
    memset(dp, 0x7F, sizeof dp);
    
    for (int j = 0; j < numsSize; ++j) {
        int num = nums[j];
        for (int i = 0; i < 3; ++i) {
            if (dp[i] < num) continue;
            if (i == 2) return true;
            dp[i] = num;
            break;
        }
    }
    
    return false;
}

File: Algorithms/[1, 1k)/334/increasing-triplet-subsequence.go
package main

import "fmt"

func increasingTriplet(nums []int) bool {
	if len(nums) < 3 {
		return false
	}

	dp := [3]int{0x7FFFFFFF, 0x7FFFFFFF, 0x7FFFFFFF}

	for _, num := range nums {
		for i, v := range dp {
			if v < num {
				continue
			}
			if i == 2 {
				return true
			}
			dp[i] = num
			break
		}
	}
	return false
}

func main() {
	fmt.Println(increasingTriplet([]int{1, 2, 3, 4, 5}))
}
